asv_bench/benchmarks/pandas/join_merge.py

```diff
@@ -0,0 +1,261 @@
+import string
+
+import numpy as np
+
+from modin.pandas import (
+    DataFrame,
+    MultiIndex,
+    Series,
+    concat,
+    date_range,
+    merge,
+)
+
+import pandas._testing as tm
+
+
+class ConcatDataFrames:
+
+    params = ([0, 1], [True, False])
+    param_names = ["axis", "ignore_index"]
+
+    def setup(self, axis, ignore_index):
+        frame_c = DataFrame(np.zeros((10000, 200), dtype=np.float32, order="C"))
+        self.frame_c = [frame_c] * 20
+        frame_f = DataFrame(np.zeros((10000, 200), dtype=np.float32, order="F"))
+        self.frame_f = [frame_f] * 20
+
+    def time_c_ordered(self, axis, ignore_index):
+        concat(self.frame_c, axis=axis, ignore_index=ignore_index)
+
+    def time_f_ordered(self, axis, ignore_index):
+        concat(self.frame_f, axis=axis, ignore_index=ignore_index)
+
+
+class Join:
+
+    params = [True, False]
+    param_names = ["sort"]
+
+    def setup(self, sort):
+        level1 = tm.makeStringIndex(10).values
+        level2 = tm.makeStringIndex(1000).values
+        codes1 = np.arange(10).repeat(1000)
+        codes2 = np.tile(np.arange(1000), 10)
+        index2 = MultiIndex(levels=[level1, level2], codes=[codes1, codes2])
+        self.df_multi = DataFrame(
+            np.random.randn(len(index2), 4), index=index2, columns=["A", "B", "C", "D"]
+        )
+
+        self.key1 = np.tile(level1.take(codes1), 10)
+        self.key2 = np.tile(level2.take(codes2), 10)
+        self.df = DataFrame(
+            {
+                "data1": np.random.randn(100000),
+                "data2": np.random.randn(100000),
+                "key1": self.key1,
+                "key2": self.key2,
+            }
+        )
+
+        self.df_key1 = DataFrame(
+            np.random.randn(len(level1), 4), index=level1, columns=["A", "B", "C", "D"]
+        )
+        self.df_key2 = DataFrame(
+            np.random.randn(len(level2), 4), index=level2, columns=["A", "B", "C", "D"]
+        )
+
+        shuf = np.arange(100000)
+        np.random.shuffle(shuf)
+        self.df_shuf = self.df.reindex(self.df.index[shuf])
+
+    def time_join_dataframe_index_multi(self, sort):
+        self.df.join(self.df_multi, on=["key1", "key2"], sort=sort)
+
+    def time_join_dataframe_index_single_key_bigger(self, sort):
+        self.df.join(self.df_key2, on="key2", sort=sort)
+
+    def time_join_dataframe_index_single_key_small(self, sort):
+        self.df.join(self.df_key1, on="key1", sort=sort)
+
+    def time_join_dataframe_index_shuffle_key_bigger_sort(self, sort):
+        self.df_shuf.join(self.df_key2, on="key2", sort=sort)
+
+    def time_join_dataframes_cross(self, sort):
+        self.df.loc[:2000].join(self.df_key1, how="cross", sort=sort)
+
+
+class JoinIndex:
```

Comment:
I'm not sure if this benchmark is any different from the previous one for Modin's implementation.